### ISSUE TITLE
Ref lookups on old values return current entity

### ISSUE THREAD

**Issue by filipesilva** [replikativ/datahike#355]
When using Datahike 0.3.6 I can see the following behaviour:

```clojure
  (let [config {:store {:backend :mem
                        :id      "default"}}
        schema [{:db/ident       :node/title
                 :db/valueType   :db.type/string
                 :db/cardinality :db.cardinality/one
                 :db/unique      :db.unique/identity}]
        conn (d/connect config)]

    (d/transact conn schema)
    (d/transact conn [{:db/id 500 :node/title "one"}])
    (d/transact conn [{:db/id 500 :node/title "two"}])
    ;; Entity 500 has title "two" now.
    (d/datoms @conn :eavt 500)                 ;; (#datahike/Datom [500 :node/title two 536870916 true])
    ;; But looking up title "one" still returns datom 500 even though that's not the title anymore.
    ;; Note the tx time 536870915, instead of 536870916.
    (d/datoms @conn :avet :node/title "one")   ;; (#datahike/Datom [500 :node/title one 536870915 true])
    ;; This leads ref lookups for a title that doesn't exist anymore to return results for the new title.
    (d/datoms @conn :eavt [:node/title "one"]) ;; (#datahike/Datom [500 :node/title two 536870916 true])
    )
```

I think the correct behaviour is that both `(d/datoms @conn :avet :node/title "one")` and `(d/datoms @conn :eavt [:node/title "one"])` return nothing instead.

cc @neotyk

**Comment by filipesilva** [comment]
@kordano mentions this is currently fixed on master (I guess via https://github.com/replikativ/datahike/pull/320). 

Current workarounds are either to go down to `0.3.2` or up to `0.3.7-SNAPSHOT`.

**Comment by filipesilva** [comment]
Fixed via #320
